compiler/hash-typecheck/src/exhaustiveness/lower.rs

```diff
@@ -135,123 +54,319 @@ impl<'tc> LowerPatOps<'tc> {
         Self { storage }
     }
 
-    /// Take a [primitives::Pat] and convert it into [Pat], essentially lowering
-    /// the pattern.
-    pub fn lower_pat(&self, id: PatId) -> Pat {
+    /// Convert a [Pat] into a [DeconstructedPat].
+    pub(crate) fn deconstruct_pat(&self, ty: TermId, pat_id: PatId) -> DeconstructedPat {
         let reader = self.reader();
+        let pat = reader.get_pat(pat_id);
+
+        let (ctor, fields) = match pat {
+            Pat::Wild
+            | Pat::Mod(_)
```

Comment:
How is mod a wildcard? Is it always exhaustive?

Reply:
Yes it is always exhaustive, you can't create a non-exhaustive `mod` pattern

Reply:
What about `{ value as Some(value) } := import("my_mod");`

Reply:
Yes, I mean it could basically be treated as a struct-like pattern, where these fields are represented as these patterns, and the rest are treated as wildcards.